### DISCUSSION TITLE
Distinguer Preview (prévisualisation), Annotate (annoter avec Hypothesis) et Sharing (partage)

### DISCUSSION THREAD

**Discussion by thom4parisot** [EcrituresNumeriques/stylo#1154] [Ideas]
Aujourd'hui, on se sert de la prévisualisation comme interface de partage (accessible au tout venant, sans protection / authentification).

Aujourd'hui, prévisualiser fait sortir de son contexte d'écriture (#1139 en est un rappel).

## Prévisualiser

La prévisualiation se passerait dans la même interface que celle d'écriture (avec les panneaux latéraux). On pourrait passer de l'un·e à l'autre. Un peu [comme là](https://github.com/EcrituresNumeriques/stylo/pull/718) (boutons "Edit" / "Preview" côte à côte) : 

![image](https://user-images.githubusercontent.com/138627/211886804-9cd65826-1e11-4474-8ecf-2916ad837dbc.png)

On peut prévisualiser si l'article est partagé avec nous.

## Niveaux de partage

On peut imaginer plusieurs niveaux de partage, qui reflètent plusieurs niveaux d'engagement : 

- lecture (accessible avec authentification, sans possibilité de changer le contenu)
- annotation (idem que lecture, annotation Hypothesis en plus)
- écriture (accessible avec authentification, même capacités que l'auteur·ice principal·e)
- publication (accessible sur le web, sans authentification)

Quand un article est partagé, on génère une **URL spécifique à ce partage**.

J'ai deux modèles de partage en tête, celui de Google Docs (une URL par document) et NextCloud (une URL par partage). On peut commencer par l'un (plus simple) et évoluer vers le second (plus de travail).

**Comment by Pixflowave** [comment]
J'aime bien la possibilité de pouvoir personnaliser l'url 
![Capture d’écran 2024-12-13 à 01 03 28](https://github.com/user-attachments/assets/f33403b9-970f-4d06-8549-93b9cc1ce21b)
